src/imp/tracer_imp.js

```diff
@@ -183,24 +185,57 @@ export default class TracerImp extends EventEmitter {
     }
 
     startSpan(fields) {
-        let spanImp = new SpanImp(this);
+        // First, assemble the SpanContextImp for our SpanImp.
+        //
+        // Normalize `fields.reference` and `fields.references` (at most one
+        // will be defined).
+        let refs = fields.references;
+        if (fields.reference) {
+            refs = [fields.reference];
+        }
+        let parCtxImp = null;
+        if (refs) {
+            for (let i = 0; i < refs.length; i++) {
+                let ref = refs[i];
+                if (ref.type() === OpenTracing.REFERENCE_CHILD_OF ||
+                        ref.type() === OpenTracing.REFERENCE_FOLLOWS_FROM) {
+                    parCtxImp = ref.spanContext().imp();
+                    break;
+                }
+            }
+        }
+        let newCtx = null;
+        if (parCtxImp === null) {
+            // Root SpanContext case:
+            newCtx = new SpanContextImp(
+                    this._platform.generateUUID(),
+                    this.generateTraceGUIDForRootSpan());
+        } else {
+            // Child SpanContextImp case:
+            newCtx = new SpanContextImp(
+                    this._platform.generateUUID(),
+                    parCtxImp._traceGUID);
+        }
+
+        let spanImp = new SpanImp(this, newCtx);
         spanImp.addTags(this._options.default_span_tags);
         spanImp.setFields(fields);
+        if (parCtxImp !== null) {
+            spanImp.setParentGUID(parCtxImp._guid);
+        }
 
         this.emit('start_trace', spanImp);
         return spanImp;
     }
 
-    inject(span, format, carrier) {
+    inject(spanContext, format, carrier) {
         switch (format) {
         case this._interface.FORMAT_TEXT_MAP:
-            this._injectToTextMap(span, carrier);
+            this._injectToTextMap(spanContext, carrier);
             break;
 
-        // The binary encoding here is optimized for correctness and uniformity
-        // across platforms: it is currently not efficient.
         case this._interface.FORMAT_BINARY:
```

Comment:
I thought it better to remove support for `FORMAT_BINARY` since it's not the same format we use in other languages... better for users to fail fast than think the inject worked when it didn't (at least not portably).